get_refs.py

```diff
@@ -0,0 +1,225 @@
+
+# genome reference files
+assembly = "Homo_sapiens.GRCh38.104"
+genome_dir = pathlib.Path("homo_sapiens/ensembl-104")
+ref_subdirs =[
+    "fasta/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz",
+    "fasta/Homo_sapiens.GRCh38.dna.primary_assembly.fa.fai",
+    "annotation/Homo_sapiens.GRCh38.104.gtf.gz",
+    "annotation/Homo_sapiens.GRCh38.104.mitogenes.txt",
+    "annotation/Homo_sapiens.GRCh38.104.spliced_intron.tx2gene_3col.tsv",
+    "annotation/Homo_sapiens.GRCh38.104.spliced_cdna.tx2gene.tsv"
+]
+ref_paths = [genome_dir / sd for sd in ref_subdirs]
+
+# salmon index files
+salmon_index_files = [
+    "complete_ref_lens.bin",
+    "ctable.bin",
+    "ctg_offsets.bin",
+    "duplicate_clusters.tsv",
+    "info.json",
+    "mphf.bin",
+    "pos.bin",
+    "pre_indexing.log",
+    "rank.bin",
+    "ref_indexing.log",
+    "refAccumLengths.bin",
+    "reflengths.bin",
+    "refseq.bin",
+    "seq.bin",
+    "versionInfo.json"
+]
+
+salmon_index_dirs = [
+    genome_dir / "salmon_index/Homo_sapiens.GRCh38.104.spliced_intron.txome",
+    genome_dir / "salmon_index/Homo_sapiens.GRCh38.104.spliced_cdna.txome"
+]
+for sa_dir in salmon_index_dirs:
+    ref_paths += [sa_dir / f for f in salmon_index_files]
+
+# star index files
+star_index_files = [
+    "chrLength.txt",
+    "chrName.txt",
+    "chrNameLength.txt",
+    "chrStart.txt",
+    "exonGeTrInfo.tab",
+    "exonInfo.tab",
+    "geneInfo.tab",
+    "Genome",
+    "genomeParameters.txt",
+    "Log.out",
+    "SA",
+    "SAindex",
+    "sjdbInfo.txt",
+    "sjdbList.fromGTF.out.tab",
+    "sjdbList.out.tab",
+    "transcriptInfo.tab"
+]
+
+star_dir = genome_dir / "star_index/Homo_sapiens.GRCh38.104.star_idx"
+if args.star_index:
+    ref_paths += [star_dir / f for f in star_index_files]
+
+# Cell Ranger index files
+cr_index_files = [
+    "reference.json",
+    "fasta/genome.fa",
+    "fasta/genome.fa.fai",
+    "genes/genes.gtf.gz",
+    "star/chrLength.txt",
+    "star/chrName.txt",
+    "star/chrNameLength.txt",
+    "star/chrStart.txt",
+    "star/exonGeTrInfo.tab",
+    "star/exonInfo.tab",
+    "star/geneInfo.tab",
+    "star/Genome",
+    "star/genomeParameters.txt",
+    "star/SA",
+    "star/SAindex",
+    "star/sjdbInfo.txt",
+    "star/sjdbList.fromGTF.out.tab",
+    "star/sjdbList.out.tab",
+    "star/transcriptInfo.tab"
+]
+cr_dir = genome_dir / "cellranger_index/Homo_sapiens.GRCh38.104_cellranger_full"
+if args.cellranger_index:
+    ref_paths += [cr_dir / f for f in cr_index_files]
+
+# barcode file paths
+barcode_dir = pathlib.Path("barcodes/10X")
+barcode_files = [
+    "3M-february-2018.txt",
+    "737K-august-2016.txt",
+    "cellranger_mit_license.txt",
+    "visium-v1.txt",
+    "visium-v2.txt"
+]
+
+ref_paths += [barcode_dir / f for f in barcode_files]
+
+## download all the files and put them in the correct locations ##
+print("Downloading reference files...")
+for path in ref_paths[0:2]:
```

Comment:
```suggestion
for path in ref_paths:
```
I don't think this should be here, otherwise it would only download the first 2

Reply:
Oh yes, that was for testing. I thought I took it out!